spec/models/post_spec.rb

```diff
@@ -0,0 +1,87 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  let(:person1) do
+    User.create(
+      name: 'Precious Betine',
+      photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+      bio: 'Student at Microverse.',
+      posts_counter: 0
+    )
+  end
+
+  subject do
+    Post.new(
+      author: person1,
+      title: 'Hello',
+      text: 'This is my first post',
+      likes_counter: 0,
+      comments_counter: 0
+    )
+  end
+
+  before do
+    subject.save
+  end
+
+  context '#title' do
+    it 'title should be present' do
+      subject.title = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'title should be less than 250 characters' do
+      subject.title = (0...300).map { ('a'..'z').to_a[rand(26)] }.join
+      expect(subject).to_not be_valid
+    end
+  end
+
+  context '#comments_counter' do
+    it 'comments counter should be greater than or equal to 0' do
+      subject.comments_counter = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'comments counter should accept only integer values' do
+      subject.comments_counter = 2.45
+      expect(subject).to_not be_valid
+    end
```

Comment:
- [ ] [OPTIONAL] You can include tests for `0` and positive integers here so your validations are fully tested